### PR TITLE
Add wait/test all/any/some API with "vector" cmp_values

### PR DESCRIPTION
**Pull request by davidozog**
As discussed during the voting process for #214 (wait/test all/any/some), user applications have a need for the new wait/test APIs to compare symmetric objects against _distinct_ values (i.e. an array of `cmp_values` instead of a single `cmp_value`).  I'm proposing a new set of "vector" wait/test routines to accomplish this, and I'm looking for feedback on this approach.  In particular, are we satisfied adding this new set of routines?  Would you prefer "v" or "_v" for the routine name instead of "_vector"? Something else perhaps?

Side note:  Our RMA working group discussions regarding the put-with-signal set/add feature suggest that we might add a return value to `shmem_wait_until` and `shmem_test`, and similarly, add another output array argument to the all/any/some APIs.  I think we can handle this incrementally; that is, we can introduce the `cmp_values` array in this proposal, then incorporate an array of signal update values in a separate proposal.  However, I'm open to alternatives.

### CODE REVIEW

File: content/shmem_test_some_vector.tex [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    satisfies the test condition at the calling \ac{PE}.  This routine does not
+    block and returns zero if no entries in \VAR{ivars} satisfied the test
+    condition.  \newtext{This routine compares each element of the \VAR{ivars}
+    array in the test set with each respective value in \VAR{cmp\_values}
+    according to the comparison operator \VAR{cmp} at the calling \ac{PE}}.
+    This routine tests all elements of \VAR{ivars} in the test set at least
+    once, and the order in which the elements are tested is unspecified.  If an
+    entry $i$ in \VAR{ivars} within the test set satisfies the test condition,
+    a series of calls to \FUNC{shmem\_test\_some\newtext{\_vector}} must
+    eventually return $i$.
```
Comment (nspark) [inline content/shmem_test_some_vector.tex]: I think the closing sentence of this paragraph is a mistaken copy-paste from elsewhere. The closing sentence of the following paragraph seems more appropriate for this particular function.

**Comment by davidozog** [comment]
Special ballot diff:
[2aa3605..4fae811](https://github.com/davidozog/openshmem-specification/compare/2aa3605..4fae811)

**Comment by jdinan** [comment]
Sync section committee, please review 2bd032d that merges changes from #267 into this PR.  I will need all section committee members to approve before this PR can be merged: @naveen-rn @nspark @akhillanger @BryantLam 

5c0d48e is a document edit.  Need approval from @davidozog and one other committee member.

**Comment by davidozog** [comment]
I approve 5c0d48e - thanks @jdinan.

**Comment by akhillanger** [comment]
@jdinan I approve 2bd032d.

**Comment by BryantLam** [comment]
@jdinan - https://github.com/openshmem-org/specification/commit/2bd032d1fbddc2e4922676026b7ccf3708a3c31b is okay. As a doc edit on this commit, I'd prefer to line break on each new sentence to minimize future diffs.

The other two doc edits look okay.

**Comment by nspark** [comment]
:+1: on 2bd032d

**Comment by naveen-rn** [comment]
+1 on 2bd032d

**Comment by davidozog** [comment]
@jdinan  This PR did not make any updates to the change log annex.  Can you remind me how you prefer to handle this?

**Comment by jdinan** [comment]
@davidozog Thanks for catching this!  Please post a PR with the changelog entry, add the SectionCommittee label and flag @BryantLam.

**Pull request merged**